From b1df3429f723ca824aa28532d55aab36e91b3e8d Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:05:23 -0400
Subject: [powerpc] scom: Create debugfs files using ibm, chip-id if available

Message-id: <1410545655-205645-95-git-send-email-dzickus@redhat.com>
Patchwork-id: 94122
O-Subject: [RHEL7 PATCH 094/626] powerpc/scom: Create debugfs files using ibm, chip-id if available
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit 762fd3ab6d2c4b438ce49f54860dc509e591209c
Author: Benjamin Herrenschmidt <benh@kernel.crashing.org>
Date:   Thu Aug 29 16:56:59 2013 +1000

    powerpc/scom: Create debugfs files using ibm,chip-id if available

    When creating the debugfs scom files, use "ibm,chip-id" as the scom%d
    index rather than a simple made up number when possible.

    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/sysdev/scom.c b/arch/powerpc/sysdev/scom.c
index 413622d..cb20d54 100644
--- a/arch/powerpc/sysdev/scom.c
+++ b/arch/powerpc/sysdev/scom.c
@@ -196,8 +196,13 @@ static int scom_debug_init(void)
   return -1;
 
  i = rc = 0;
- for_each_node_with_property(dn, "scom-controller")
-  rc |= scom_debug_init_one(root, dn, i++);
+ for_each_node_with_property(dn, "scom-controller") {
+  int id = of_get_ibm_chip_id(dn);
+  if (id == -1)
+   id = i;
+  rc |= scom_debug_init_one(root, dn, id);
+  i++;
+ }
 
  return rc;
 }
-- 
1.7.1